google-cloud-spanner/src/main/java/com/google/cloud/spanner/TransactionRunnerImpl.java

```diff
@@ -698,6 +712,13 @@ public void run() {
       return updateCounts;
     }
 
+    @Override
+    public void withTransactionTag(String tag) {
+      synchronized (lock) {
```

Comment:
I think it would be good to have a check here that this method may only be called before `TransactionRunner` has been executed.